src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/dataset-settings-webhooks-tab.component.ts

```diff
@@ -0,0 +1,216 @@
+    public dataSource = new MatTableDataSource();
+
+    public isLoaded: boolean = false;
+    public readonly DISPLAY_COLUMNS: string[] = ["event", "status", "actions"];
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+
+    private ngbModalService = inject(NgbModal);
+    private datasetWebhooksService = inject(DatasetWebhooksService);
+    private cdr = inject(ChangeDetectorRef);
+    private modalService = inject(ModalService);
+
+    public ngOnInit(): void {
+        this.updateTable();
+    }
+
+    private updateTable(): void {
+        this.datasetWebhooksService
+            .datasetWebhookSubscriptions(this.datasetBasics.id)
+            .pipe(
+                finalize(() => {
+                    this.isLoaded = true;
+                    this.cdr.detectChanges();
+                }),
+                takeUntilDestroyed(this.destroyRef),
+            )
+
+            .subscribe((data: WebhookSubscription[]) => {
+                this.dataSource.data = data;
+                this.cdr.detectChanges();
+            });
+    }
+
+    public get datasetBasics(): DatasetBasicsFragment {
+        return this.webhooksViewData.datasetBasics;
+    }
+
+    public createWebhook(): void {
+        const modalRef = this.ngbModalService.open(CreateEditSubscriptionModalComponent);
+        const modalRefInstance = modalRef.componentInstance as CreateEditSubscriptionModalComponent;
+        modalRefInstance.datasetBasics = this.webhooksViewData.datasetBasics;
+
+        from(modalRef.result)
+            .pipe(
+                switchMap((result: WebhookSubscriptionModalActionResult) =>
+                    result.payload
+                        ? this.datasetWebhooksService.datasetWebhookCreateSubscription(
+                              this.datasetBasics.id,
+                              result.payload,
+                          )
+                        : of(),
+                ),
+                takeUntilDestroyed(this.destroyRef),
+            )
+            .subscribe((data: CreateWebhookSubscriptionSuccess | null) => {
+                if (data) {
+                    const modalRef = this.ngbModalService.open(CreateEditSubscriptionModalComponent, {
+                        backdrop: "static",
+                        keyboard: false,
+                    });
+                    const modalRefInstance = modalRef.componentInstance as CreateEditSubscriptionModalComponent;
+                    modalRefInstance.datasetBasics = this.webhooksViewData.datasetBasics;
+                    modalRefInstance.subscriptionData = data;
+
+                    from(modalRef.result)
+                        .pipe(
+                            switchMap((result: WebhookSubscriptionModalActionResult) => {
+                                if (result.action === WebhookSubscriptionModalAction.CLOSE) {
+                                    this.updateTable();
```

Comment:
Same here, why do we need to reload the entire table, when we know we've just added 1 row.